main.js

```diff
@@ -65,6 +66,44 @@ function getCaptureError (client, _captureError) {
 	};
 }
 
+function sanitiseError (err) {
+	//Initially check for a specific email field e.g. in a url in an attempt to not blitz helpful error information if possible
+	if (err.message) {
+		const origErrorMsg = err.message
+		const emailFieldRegex = 'email='
+		const indxEmailField = origErrorMsg.search(emailFieldRegex)
+
+		let cleanedErr = origErrorMsg
+		if (indxEmailField > 0) {
+
+			cleanedErr = origErrorMsg.slice(0,indxEmailField) + 'email=-redacted-'
+
+			//in case there are more fields / is further at the end of the string
+			let indexOfAdditionalParams = origErrorMsg.indexOf('&', indxEmailField)
+			let indexOfPostURLspace = origErrorMsg.indexOf(' ', indxEmailField)
+
+			if(indexOfAdditionalParams > 0) {
+				cleanedErr = cleanedErr + origErrorMsg.slice(indexOfAdditionalParams, err.length)
+			}
+			else if (indexOfPostURLspace > 0){
+				cleanedErr = cleanedErr + origErrorMsg.slice(indexOfPostURLspace, err.length)
+			}
+
+		}
+
+		//The catch all check in case there are any remaining email addresses in the error
+		const emailRegex = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/g;
```

Comment:
/me has a flashback of that giphy about the 0.01% of 99.99% ... small but a very thorny one! :) I think this is ok, along with the other measures we're putting in place.